examples/applications/plot_cyclical_feature_engineering.py

```diff
@@ -0,0 +1,765 @@
+# on a range of held out data points.
+naive_linear_pipeline.fit(X.iloc[train_0], y.iloc[train_0])
+naive_linear_predictions = naive_linear_pipeline.predict(X.iloc[test_0])
+
+one_hot_linear_pipeline.fit(X.iloc[train_0], y.iloc[train_0])
+one_hot_linear_predictions = one_hot_linear_pipeline.predict(X.iloc[test_0])
+
+cyclic_cossin_linear_pipeline.fit(X.iloc[train_0], y.iloc[train_0])
+cyclic_cossin_linear_predictions = cyclic_cossin_linear_pipeline.predict(
+    X.iloc[test_0]
+)
+
+cyclic_spline_linear_pipeline.fit(X.iloc[train_0], y.iloc[train_0])
+cyclic_spline_linear_predictions = cyclic_spline_linear_pipeline.predict(
+    X.iloc[test_0]
+)
+
+# %%
+# We visualize those predictions by zooming on the last 96 hours (4 days) of
+# the test set to get some qualitative insights:
+last_hours = slice(-96, None)
+fig, ax = plt.subplots(figsize=(12, 4))
+fig.suptitle("Predictions by linear models")
+ax.plot(
+    y.iloc[test_0].values[last_hours],
+    "x-",
+    alpha=0.2,
+    label="Actual demand",
+    color="black",
+)
+ax.plot(naive_linear_predictions[last_hours], "x-", label="Ordinal time features")
+ax.plot(
+    cyclic_cossin_linear_predictions[last_hours],
+    "x-",
+    label="Trigonometric time features",
+)
+ax.plot(
+    cyclic_spline_linear_predictions[last_hours],
+    "x-",
+    label="Spline-based time features",
+)
+ax.plot(
+    one_hot_linear_predictions[last_hours],
+    "x-",
+    label="One-hot time features",
+)
+_ = ax.legend()
+
+# %%
+# We can draw the following conclusions from the above plot:
+#
+# - the **raw ordinal time-related features** are problematic because they do
+#   not capture the natural periodicity: we observe a big jump in the
+#   predictions at the end of each day when the hour features goes from 23 back
+#   to 0. We can expect similar artifacts at the end of each week or each year.
+#
+# - as expected, the **trigonometric features** (sine and cosine) do not have
+#   these discontinuities at midnight but the linear regression model fails to
+#   properly leverage those features to properly model intra-day variations.
+#   Using trigonometric features for higher harmonics or additional
+#   trigonometric features for the natural period with different phases could
+#   potentially fix this problem.
+#
+# - the **periodic spline-based features** fix those two problems at once: they
+#   give more expressivity to the linear model by making it possible to focus
+#   on a specific hours thanks to the use of 12 splines. Furthermore the
+#   `extrapolation="periodic"` option enforces a smooth representation between
+#   `hour=23` and `hour=0`.
+#
+# - the **one-hot encoded features** behave similarly to the periodic
+#   spline-based features but are more spiky: for instance they can better
+#   model the morning peak during the week days since this peak last fewer than
+#   an hour. However, we will see in the following that what can be an
```

Comment:
```suggestion
#   model the morning peak during the week days since this peak lasts shorter
#   than an hour. However, we will see in the following that what can be an
```